server/src/main/java/dev/slimevr/osc/VRCOSCHandler.java

```diff
@@ -236,14 +238,24 @@ public void update() {
 					}
 
 					// Send regular trackers' rotations
-					var unity = computedTrackers
-						.get(i)
-						.getRotation()
-						.toEulerAngles(EulerOrder.ZXY);
+					Quaternion rot = computedTrackers.get(i).getRotation();
+					// We flip the X and Y components of the quaternion because
+					// we flip the z direction when communicating from
+					// our right-handed API to VRChat's left-handed API.
+					// X quaternion represents a rotation from y to z
+					// Y quaternion represents a rotation from z to x
+					// When we negate the z direction, X and Y quaternion
+					// components must be negated.
+					EulerAngles unityAngles = new Quaternion(
+						rot.getW(),
+						-rot.getX(),
+						-rot.getY(),
+						rot.getZ()
+					).toEulerAngles(EulerOrder.YXZ);
```

Comment:
This... Seems really weird? Shouldn't the euler angles be the same no matter what as long as the order is correct?

Reply:
No, because euler angle standards are all over the place even with the same order according to AxisAngle.

Reply:
That's stupid, I hate Euler angles. =w=
Make sure this is tested thoroughly, if it works and seems to be necessary, then that's fine with me.

Reply:
I went through it with AxisAngle and tested it yes ^^